pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -82,48 +76,20 @@ func NewIstioStatusUpdater(instance *v1alpha1.IstioControlPlane) helmreconciler.
 
 // EndReconcile updates the status field on the IstioControlPlane instance based on the resulting err parameter.
 func (u *IstioStatusUpdater) EndReconcile(_ runtime.Object, err error) error {
+	// TODO: status should track actual component status.
 	status := u.instance.Status
+	vstatus := &v1alpha2.InstallStatus_VersionStatus{}
 	if err == nil {
-		if condition := status.GetCondition(v1alpha1.ConditionTypeInstalled); condition.Status != v1alpha1.ConditionStatusTrue {
-			status.SetCondition(v1alpha1.Condition{
-				Type:   v1alpha1.ConditionTypeInstalled,
-				Reason: v1alpha1.ConditionReasonInstallSuccessful,
-				Status: v1alpha1.ConditionStatusTrue,
-			})
-			status.SetCondition(v1alpha1.Condition{
-				Type:   v1alpha1.ConditionTypeReconciled,
-				Reason: v1alpha1.ConditionReasonInstallSuccessful,
-				Status: v1alpha1.ConditionStatusTrue,
-			})
-		} else {
-			status.SetCondition(v1alpha1.Condition{
-				Type:   v1alpha1.ConditionTypeReconciled,
-				Reason: v1alpha1.ConditionReasonReconcileSuccessful,
-				Status: v1alpha1.ConditionStatusTrue,
-			})
-		}
+		vstatus.Status = v1alpha2.InstallStatus_HEALTHY
 	} else {
-		if condition := status.GetCondition(v1alpha1.ConditionTypeInstalled); condition.Status != v1alpha1.ConditionStatusTrue {
-			status.SetCondition(v1alpha1.Condition{
-				Type:    v1alpha1.ConditionTypeInstalled,
-				Reason:  v1alpha1.ConditionReasonInstallError,
-				Status:  v1alpha1.ConditionStatusFalse,
-				Message: fmt.Sprintf("errors occurred during installation: %s", err),
-			})
-			status.SetCondition(v1alpha1.Condition{
-				Type:   v1alpha1.ConditionTypeReconciled,
-				Reason: v1alpha1.ConditionReasonInstallError,
-				Status: v1alpha1.ConditionStatusFalse,
-			})
-		} else {
-			status.SetCondition(v1alpha1.Condition{
-				Type:    v1alpha1.ConditionTypeReconciled,
-				Reason:  v1alpha1.ConditionReasonReconcileError,
-				Status:  v1alpha1.ConditionStatusFalse,
-				Message: fmt.Sprintf("errors occurred during reconciliation: %s", err),
-			})
-		}
+		vstatus.Status = v1alpha2.InstallStatus_ERROR
 	}
+	status.TrafficManagement = vstatus
+	status.ConfigManagement = vstatus
+	status.PolicyTelemetry = vstatus
+	status.Security = vstatus
+	status.IngressGateway = vstatus
+	status.EgressGateway = vstatus
```

Comment:
This could be error prune because I may not ask for installing policy telemetry and I'm getting status Healthy because my pilot is healthy.

Reply:
is this change added or still TODO